readers/common/src/main/java/com/criteo/hadoop/garmadon/reader/helper/ReaderUtils.java

```diff
@@ -0,0 +1,39 @@
+package com.criteo.hadoop.garmadon.reader.helper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReaderUtils {
+    public static final Runnable EMPTY_ACTION = () -> {
+    };
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReaderUtils.class);
+
+    protected ReaderUtils() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static boolean retryAction(Runnable action, String exceptionStr, Runnable actionFailure) {
+        final int maxAttempts = 5;
+
+        for (int retry = 1; retry <= maxAttempts; ++retry) {
+            try {
+                action.run();
```

Comment:
use Callable for action. That allows to return what is potentially generated by the action (cna serve more use cases than Runnable and return boolean) + Callable can throw Exception. So you don't have to catch IOException and wrap in RuntimeException for instance